helix-term/src/ui/picker.rs

```diff
@@ -413,9 +424,33 @@ impl<T: Item> Picker<T> {
 
         log::debug!("picker score {:?}", Instant::now().duration_since(now));
 
-        // reset cursor position
-        self.cursor = 0;
-        self.previous_pattern.clone_from(pattern);
+        // reset cursor position or recover position based on previous matched option
+        if let Some(prev_matched_option) = prev_matched_option {
+            self.cursor = self
+                .matches
+                .iter()
+                .enumerate()
+                .find_map(|(index, m)| {
+                    if m.0 == prev_matched_option {
+                        Some(index)
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or(0);
+        }
+        self.previous_pattern = pattern;
+    }
+
+    pub fn score(&mut self, reset_cursor: bool) {
+        let pattern = self.prompt.line().clone();
+        self.recalculate_score(Some(pattern), reset_cursor);
+    }
+
+    pub fn add_options(&mut self, mut options: Vec<T>) {
+        self.options.append(&mut options);
+        self.dirty = true;
```

Comment:
It doesn't seemed to reset to `false`? But yeah, passing it in `recalculate_score` is better.

Reply:
resetted it to the old version...